fiware-rss/src/main/java/es/upm/fiware/rss/ws/SettlementService.java

```diff
@@ -85,29 +83,63 @@ public Response launchSettlement(SettlementJob task) throws Exception {
             String[] args = {"callbackUrl"};
             throw new RSSException(UNICAExceptionType.CONTENT_NOT_WELL_FORMED, args);
         }
-        
+
         // Launch process
         settlementManager.runSettlement(task);
         Response.ResponseBuilder rb = Response.status(Response.Status.ACCEPTED.getStatusCode());
         return rb.build();
     }
 
+    private boolean applyPatch(int id, PatchAction action) {
+        // If we need more patchs add them here, by now only replace paids
+        if (action.getPath().equals("/paid") && action.getOp().equals("replace")) {
+            return settlementManager.setPayReport(id, action.getValue().equals("true")).orElse(false);
+        }
+
+        return false;
+    }
+
+    /**
+     * PATCH based on RFC5789, where it receives a list of description of changes.
+     * We get a list of descriptions like {"op": "replace", "path": "/parameter", "value": "value"}
+     * To read more about it: http://williamdurand.fr/2014/02/14/please-do-not-patch-like-an-idiot/
+     */
+    @WebMethod
+    @PATCH
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Path("/reports/{id}")
+    public Response patchReport(@PathParam("id") int id, List<PatchAction> actions) throws Exception {
+        Response.ResponseBuilder rb = Response.status(Response.Status.OK.getStatusCode());
+        JsonResponse response = new JsonResponse();
+
+        if (this.userManager.isAdmin()) {
+            // Apply actions (First map needs lambda function because lack of partial application
+            response.setSuccess(actions.stream().map(act -> this.applyPatch(id, act)).allMatch(Boolean::booleanValue));
+        }
+        rb.entity(response);
+        return rb.build();
+    }
+
     @WebMethod
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     @Path("/reports")
     public Response getReports(
             @QueryParam("aggregatorId") String aggregatorId,
             @QueryParam("providerId") String providerId,
-            @QueryParam("productClass") String productClass)
+            @QueryParam("productClass") String productClass,
+            @DefaultValue("true") @QueryParam("all") boolean all,
```

Comment:
I don't think that you need an specific query string for retrieving all the reports. Normally, when you don't include the pagination query strings means that you don't want a page so all of them are returned. Indeed with your implementation this actually works since the default values are 0 and -1 (from the first all of them)